includes/class-wc-payments.php

```diff
@@ -841,4 +843,44 @@ function( string $cookie_hash ) {
 
 		return (int) $_SERVER['HTTP_X_WCPAY_PLATFORM_CHECKOUT_USER'];
 	}
+
+	/**
+	 * Remove default billing email field for Platform Checkout
+	 *
+	 * @param array $fields WooCommerce checkout fields.
+	 * @return array WooCommerce checkout fields.
+	 */
+	public static function platform_checkout_remove_default_email_field( $fields ) {
+		unset( $fields['billing']['billing_email'] );
```

Comment:
Something like this should do it.
```suggestion
		$wc_checkout = WC_Checkout::instance();
		$default_fields = (function () {
			return $this->fields;
		})->bindTo( $wc_checkout, WC_Checkout )();
		if ( $default_fields === $fields ) {
			unset( $fields['billing']['billing_email'] );
		} else {
			remove_action( 'woocommerce_checkout_before_customer_details', [ __CLASS__, 'platform_checkout_fields_before_billing_details' ], 20 );
		}
```